bin/varnishd/cache/cache_backend_probe.c

```diff
@@ -444,6 +444,37 @@ vbp_heap_insert(struct vbp_target *vt)
 /*--------------------------------------------------------------------
  */
 
+/*
+ * called when a task was successful or could not get scheduled
+ * returns non-NULL if target is to be deleted (outside mtx)
+ */
+static struct vbp_target *
+vbp_task_complete(struct vbp_target *vt)
+{
+	CHECK_OBJ_NOTNULL(vt, VBP_TARGET_MAGIC);
+
+	Lck_AssertHeld(&vbp_mtx);
+
+	assert(vt->heap_idx == VBH_NOIDX);
+
+	if (vt->state == vbp_state_scheduled) {
+		WRONG("vbp_state_scheduled");
+	} else if (vt->state == vbp_state_running) {
+		vt->state = vbp_state_scheduled;
+		vt->due = VTIM_real() + vt->interval;
+		vbp_heap_insert(vt);
+		vt = NULL;
+	} else if (vt->state == vbp_state_cold) {
+		WRONG("vbp_state_cold");
+	} else if (vt->state == vbp_state_cooling) {
+		vt->state = vbp_state_cold;
+		vt = NULL;
+	} else {
+		assert(vt->state == vbp_state_deleted);
+	}
```

Comment:
Why not fold both `WRONG()` statements in the else clause?

```c
if (vt->state == vbp_state_running) {
        /* ... */
} else if (vt->state == vbp_state_cooling) {
        /* ... */
} else if (vt->state != vbp_state_deleted)
        WRONG(vt->state->name);
```

Reply:
because I did not think of this way.